src/Xpress.jl

```diff
@@ -30,12 +30,40 @@ include("helper.jl")
 include("api.jl")
 include("license.jl")
 
-function initialize()
+"""
+    initialize(;
+        liccheck::Function = identity,
+        verbose::Bool = true,
+        xpauth_path::String = ""
+    )
+
+Performs license checking with `liccheck` validation function on dir
+`xpauth_path`.
+
+This function must be called before any XPRS functions can be called.
+
+By default, `__init__` calls this with no keyword arguments.
+
+## Example
+
+```julia
+ENV["XPRESS_JL_NO_AUTO_INIT"] = true
+using Xpress
+liccheck(x::Vector{Cint}) = Cint[xor(x[1], 0x0123)]
+Xpress.initialize(;
+    liccheck = liccheck,
+    verbose = false,
+    xpauth_path = "/tmp/xpauth.xpr,
+)
+# Now you can use Xpress
+```
+"""
+function initialize(; kwargs...)
```

Comment:
@joaquimg any reason to call `userlic` instead of this `initialize`? We'll keep the old behavior, but this might be simpler. No need for `dlopen` and `XPRSinit`.

Reply:
I have had to load some libraries explicitly in some cases. For instance, xprl.dll on windows.

Reply:
Hmm. We should be able to fix that. The CI binaries work find? I think it should probably also work without the explicit `dlopen`. `ccall` should do it.

Reply:
I can simply start using this new reasonable function and report back

Reply:
So I removed the `dlopen` and CI still passing. So let me know

Reply:
how does it find the dll with no dlopen?

Reply:
the jll is figuring that out?

Reply:
it would be good to document how to swap versions with he jll.
It is very common that we need specific versions to reproduce and debug

Reply:
> how does it find the dll with no dlopen?

`libxprs` is the full path, and that should be able to open any dependencies.

> it would be good to document how to swap versions with he jll.

This PR is unrelated to the jll?

Reply:
This is working on my naive tests. Will test in production during the week!

Reply:
Since we're going to need to do a bunch of testing before tagging the next release (which I'll make as breaking, just to be on the safe side), objections to merging this now so you can test on `master` or with the `Xpress_jll` patch?

Reply:
Let's wait Monday night. Today is Sunday in Brazil :)